modules/explorer-api/src/main/scala/org/ergoplatform/explorer/http/api/v1/services/Mempool.scala

```diff
@@ -67,6 +75,39 @@ object Mempool {
 
     import repo._
 
+    def hasUnconfirmedBalance(ergoTree: ErgoTree): F[Boolean] =
+      txs
+        .countByErgoTree(ergoTree.value)
+        .map(_ > 0)
+        .thrushK(trans.xa)
+
+    def getUnconfirmedBalanceByAddress(
+      address: Address,
+      confirmedBalance: Balance
+    ): F[TotalBalance] = {
```

Comment:
@monyedavid Why making one service method relying on another?
`confirmedBalanceOf(addr) >>=  getUnconfirmedBalanceByAddress(addr, _)`
Why not making both methods independently compute their part of `TotalBalance` and then construct TotalBalance from those parts in a separate `TotalBalance.apply(confirmed, unconfirmed)` smart constructor?